packages/react-composites/src/composites/CallComposite/components/CallControls.tsx

```diff
@@ -89,41 +98,43 @@ export const CallControls = (props: CallControlsProps & ContainerRectProps): JSX
     [props.isMobile, props.options]
   );
 
-  const adapter = useAdapter();
-
   const localeStrings = useLocale();
 
   /* @conditional-compile-remove(DNS) */
   const [isDeepNoiseSuppressionOn, setDeepNoiseSuppressionOn] = useState<boolean>(false);
 
+  /* @conditional-compile-remove(DNS) */
+  const adapter = useAdapter();
   /* @conditional-compile-remove(DNS) */
   const startDeepNoiseSuppression = useCallback(async () => {
     await adapter.startNoiseSuppressionEffect();
   }, [adapter]);
 
+  /* @conditional-compile-remove(DNS) */
+  const deepNoiseSuppresionEffectsDependency = useSelector(getDeepNoiseSuppresionEffectsDependency);
+  /* @conditional-compile-remove(DNS) */
+  const deepNoiseSuppressionOnByDefault = useSelector(getDeepNoiseSuppresionIsOnByDefault);
   /* @conditional-compile-remove(DNS) */
   useEffect(() => {
-    if (
-      adapter.getState().onResolveDeepNoiseSuppressionDependency &&
-      adapter.getState().deepNoiseSuppressionOnByDefault
-    ) {
+    if (deepNoiseSuppresionEffectsDependency && deepNoiseSuppressionOnByDefault) {
       startDeepNoiseSuppression();
       setDeepNoiseSuppressionOn(true);
     }
-  }, [adapter, startDeepNoiseSuppression]);
+  }, [adapter, deepNoiseSuppresionEffectsDependency, deepNoiseSuppressionOnByDefault, startDeepNoiseSuppression]);
 
   /* @conditional-compile-remove(DNS) */
-  const environmentInfo = adapter.getState().environmentInfo;
+  const environmentInfo = useSelector(getEnvironmentInfo);
 
   /* @conditional-compile-remove(DNS) */
   const isSafari = _isSafari(environmentInfo);
   /* @conditional-compile-remove(DNS) */
-  const showNoiseSuppressionButton =
-    adapter.getState().onResolveDeepNoiseSuppressionDependency &&
-    !adapter.getState().hideDeepNoiseSuppressionButton &&
+  const hideDeepNoiseSuppressionButton = useSelector(getHideDeepNoiseSupressionButton);
+  /* @conditional-compile-remove(DNS) */
+  const showNoiseSuppressionButton = !!(
```

Comment:
`useMemo` might be useful here

Reply:
Booleans (and any trivial values) are OK to not memoize as they're super speedy to compute (faster than a useMemo lookup) and they don't have the problem of having an object reference compared like complex objects do

Reply:
I see, thanks for sharing! 
If you have any good docs about this to share, please do!

Reply:
great, thanks! It does sound logical that the func call will be more expensive that the value variable usage here